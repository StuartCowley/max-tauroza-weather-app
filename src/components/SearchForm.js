import React from "react";

const SearchForm = ({ searchText, setSearchText, onSubmit }) => {
  const handleInputChange = (event) => {
    setSearchText(event.target.value);
  };
  return (
    <div className="search-form">
      <label htmlFor="location-search">
        Change location
        <input
          type="text"
          placeholder="Enter a town or city"
          id="location-search"
          onChange={handleInputChange}
          value={searchText}
        />
        <button type="submit" onClick={onSubmit}>
          Search
        </button>
      </label>
    </div>
  );
};

export default SearchForm;
