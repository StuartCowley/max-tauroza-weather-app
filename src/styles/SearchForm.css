@import url("https://fonts.googleapis.com/css2?family=Goldman&family=Press+Start+2P&display=swap");

.SearchForm {
  padding: 20px;
  display: flex;
  justify-content: space-between;
}

/* label {
  font-size: 20px;
  font-weight: bold;
} */

button {
  font-family: "Press Start 2P";
  padding: 5px;
  border-radius: 5%;
}

input {
  font-family: "Press Start 2P";
  margin: 20px;
  padding: 5px;
}
