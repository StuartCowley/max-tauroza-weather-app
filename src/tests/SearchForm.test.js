import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText=""
        setSearchText={jest.fn()}
        onSubmit={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("has a search button", () => {
    const { getByRole } = render(
      <SearchForm
        searchText=""
        setSearchText={jest.fn()}
        onSubmit={jest.fn()}
      />
    );

    expect(getByRole("button")).toBeInTheDocument();
  });
});
