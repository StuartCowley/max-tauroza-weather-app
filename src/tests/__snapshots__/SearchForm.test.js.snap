// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <label
      for="location-search"
    >
      Change location
      <input
        id="location-search"
        placeholder="Enter a town or city"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </label>
  </div>
</DocumentFragment>
`;
